tools/trap/trap.c

```diff
@@ -288,6 +294,67 @@ static int extfsdump_recv(int dev_fd)
         return 0;
 }
 
+int dump_file_from_fs(int dev_fd, char *filename, int maxnamelen)
+{
+	int ret;
+	int fd;
+	int i, last_subdir;
+	char buf;
+	FILE *fp;
+	u_int64_t file_size;
+	char char_filesize[8];
+	char local_filename[maxnamelen + 9];
+
+	printf("Attempting to Dump File %s from Target Device\n", filename);
+
+	ret = write(dev_fd, filename, maxnamelen);
+	if (ret != maxnamelen) {
+		printf("Unable to send Dump File name to Target Device, ret = %d\n", ret);
+		return -1;
+	}
+
+	file_size = 0;
+	for (int i = 0; i < 8; i++) {
+		ret = read(dev_fd, &char_filesize[i], sizeof(char_filesize[i]));
+		if (ret != sizeof(char_filesize[i])) {
+			printf("Unable to get file size from Target Device, ret = %d\n", ret);
+			return -1;
+		}
+	}
+
+	file_size = (((u_int64_t)char_filesize[0] << 56) & 0xFF00000000000000U) | (((u_int64_t)char_filesize[1] << 48) & 0x00FF000000000000U) | (((u_int64_t)char_filesize[2] << 40) & 0x0000FF0000000000U) | (((u_int64_t)char_filesize[3] << 32) & 0x000000FF00000000U) | ((char_filesize[4] << 24) & 0x00000000FF000000U) | ((char_filesize[5] << 16) & 0x0000000000FF0000U) | ((char_filesize[6] <<  8) & 0x000000000000FF00U) | (char_filesize[7] & 0x00000000000000FFU);
```

Comment:
This must work. Instead of having such long statement.
```
for (int i = 0; i < 8; i++)
    filesize |= (u_int64_t)char_filesize[8 - i - 1] << (i * 8);
```

Reply:
Yes will consider this, thankyou